lib/pbench/agent/results.py

```diff
@@ -90,70 +101,94 @@ def make_result_tb(self):
-            self.logger.error("md5sum failed for %s, skipping" % tarball)
-            if os.path.exists(tarball):
-                os.remove(tarball)
-            if os.path.exists(tarball_md5):
-                os.remove(tarball_md5)
+            self.logger.error("md5sum failed for {}, skipping", tarball)
+            try:
+                tarball.unlink()
+            except OSError as exc:
+                if exc.errno != errno.ENOENT:
+                    self.logger.error("error removing failed tar ball, {}", tarball)
+            try:
+                tarball_md5.unlink()
+            except OSError as exc:
+                if exc.errno != errno.ENOENT:
+                    self.logger.error(
+                        "error removing failed tar ball MD5, {}", tarball_md5
+                    )
             sys.exit(1)
 
 
 class CopyResultTb:
-    def __init__(self, tarball, path, _logger=None):
-        config = AgentConfig()
-        server_rest_url = config.results.get("server_rest_url")
+    chunk_size = 4096
 
-        self.tarball = tarball
-        self.path = path
-        self.chunk_size = 4096
-        self.logger = logger if not _logger else _logger
+    def __init__(self, tarball, config, logger):
+        if not os.path.exists(tarball):
+            logger.error("tarball does not exist, '{}'", tarball)
+            sys.exit(1)
+        tarball_md5 = f"{tarball}.md5"
+        if not os.path.exists(tarball_md5):
+            logger.error("tarball's .md5 does not exist, '{}'", tarball_md5)
+            sys.exit(1)
+        self.tarball = Path(tarball)
+        self.tarball_md5 = Path(tarball_md5)
+        self.logger = logger
+        server_rest_url = config.results.get("server_rest_url")
         self.upload_url = f"{server_rest_url}/upload"
-        self.controller_dir = os.path.dirname(self.tarball)
-        self.controller = os.path.basename(self.controller_dir)
 
     def read_in_chunks(self, file_object):
-        while True:
-            data = file_object.read(self.chunk_size)
-            if not data:
-                break
+        data = file_object.read(self.chunk_size)
+        while data:
             yield data
+            data = file_object.read(self.chunk_size)
+
+    def copy_result_tb(self):
+        files = [f for f in self.tarball.parent.iterdir() if f.is_file()]
+        file_count = len(files)
+        if file_count != 2:
+            self.logger.error(
+                "(internal): unexpected file count, {}, associated with tarball, '{}'",
+                file_count,
+                self.tarball,
+            )
+            sys.exit(1)
 
-    def post_file(self, file):
-        filename = secure_filename(os.path.join(self.path, file))
-        with open(f"{os.path.join(self.path, file)}.md5.check", "r") as _check:
-            md5sum = _check.read()
+        with open(self.tarball_md5, "r") as md5fp:
+            md5sum = md5fp.read()
+        filename = secure_filename(str(self.tarball))
```

Comment:
What does this protect us from?

Reply:
This came from the original code, but near as I can tell, `secure_filename()` munges a full POSIX file name path into a single file name so that the `PUT` request can't cause the server to access an unwanted path.

For example, if the tar ball path is `/var/tmp/send/tarball.tar.xz`, the "secure" filename sent over the wire in the PUT headers would be `var_tmp_send_tarball.tar.xz`.